pennylane/hf/convert.py

```diff
@@ -0,0 +1,318 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the functions needed for computing the molecular Hamiltonian.
+"""
+import numpy as np
+import openfermion
+import pennylane as qml
+from openfermion import QubitOperator
+from pennylane import Hamiltonian
+from pennylane.wires import Wires
+
+
+def _process_wires(wires, n_wires=None):
+    r"""Checks and consolidates custom user wire mapping into a consistent, direction-free, ``Wires``
+    format. Used for converting between OpenFermion qubit numbering and PennyLane wire labels.
+
+    Since OpenFermion's qubit numbering is always consecutive int, simple iterable types such as
+    list, tuple, or Wires can be used to specify the 2-way `qubit index` <-> `wire label` mapping
+    with indices representing qubits. Dict can also be used as a mapping, but does not provide any
```

Comment:
It is already supported I think, see examples 4, 5. Is that right @zeyueN?

Reply:
Yes dicts are already supported, but I guess @ixfoduap 's point is more about if it's really bringing any real benefits to the table. I had similar doubt back when coding this but decided to keep it. The thinking was that it could be the case some user may not want to change the label for any wire except for a few. In this case, using a dict containing only the wire mapping one wants to change is perhaps more convenient, compared to providing a complete list of wire mapping (most of which are redundant).